spring-boot-project/spring-boot-docs/src/docs/asciidoc/features/testing.adoc

```diff
@@ -938,6 +938,9 @@ This is done by automatically defining a `Neo4jConnectionDetails` bean which is
 
 NOTE: You'll need to add the `spring-boot-testcontainers` module as a test dependency in order to use service connections with Testcontainers.
 
+NOTE: You'll need to add `@ImportAutoConfiguration(ServiceConnectionAutoConfiguration.class)` if you are using <<features#features.testing.spring-boot-applications.autoconfigured-tests, test slices>>.
```

Comment:
I've been using `@ServiceConnection` along with `@JdbcTest` without adding this. See https://github.com/eddumelendez/testcontainers-samples/blob/spring-boot-3.1.0-RC1/spring-boot-cockroachdb-flyway/src/test/java/com/example/springbootcockroachdbflyway/CockroachDbTest.java

Reply:
@eddumelendez If you define it as `@Bean` managed by spring not `@Container` field managed by testcontainers, then you need `@ImportAutoConfiguration(ServiceConnectionAutoConfiguration.class)`.

Reply:
I've amend the commit.

Reply:
Just out of curiosity why would you do that given

```
@Container
@ServiceConnection
static final GenericContainer<?> ...
```

would already work for your existing tests, without adding any additional configuration.

Reply:
I'd like to let spring manage it's lifecycle if possible, I'm doing this before `spring-boot-testcontainers` existed.

```java
abstract class JdbcDatabase<T extends JdbcDatabaseContainer<?>> {

	private final Class<T> containerClass;

	@SuppressWarnings("unchecked")
	JdbcDatabase() {
		this.containerClass = (Class<T>) ResolvableType.forClass(getClass()).as(JdbcDatabase.class).resolveGeneric(0);
	}

	protected String getImageName() {
		return getClass().getSimpleName().toLowerCase();
	}

	protected String getImageTag() {
		return "latest";
	}

	protected String getImage() {
		return getImageName() + ':' + getImageTag();
	}

	@Bean(initMethod = "start")
	T container() throws Exception {
		return this.containerClass.getConstructor(String.class).newInstance(getImage());
	}

	@Bean
	DataSource dataSource(T container) {
		HikariDataSource ds = new HikariDataSource();
		ds.setJdbcUrl(container.getJdbcUrl());
		ds.setUsername(container.getUsername());
		ds.setPassword(container.getPassword());
		return ds;
	}

}

@TestConfiguration(proxyBeanMethods = false)
public class MySQL extends JdbcDatabase<MySQLContainer<?>> {

}

```